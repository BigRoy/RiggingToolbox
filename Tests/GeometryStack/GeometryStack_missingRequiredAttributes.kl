
require RiggingToolbox;

operator entry(){
   	
   	GeometryStack stack();
   	stack.addGeometryOperator(PolygonMeshSphereGenerator(2.0, 2, true, false));
   	stack.addGeometryOperator(PushDeformer());
   	stack.initCachePoints();

    EvalContext context();
    // Causes and erorr because the PushDeformer requires normals, but the sphere doesn't have them. 
    stack.evaluate(context);
}

