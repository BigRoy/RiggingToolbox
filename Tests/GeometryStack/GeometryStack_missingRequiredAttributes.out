Error: Geometry missing required attributes:["normals"]
  1 kl.internal.String.SetErrorDataPtrAndLength.AS0()
  2 function.setError.in_AS0.ST()
  3 method.evaluate.io_AS0.UO_GeometryStack.in_AS0.OO_EvalContext() GeometryStack.kl:122
  4 operator.entry() GeometryStack_missingRequiredAttributes.kl:13
  5 kl.internal.entry.stub.cpu()
Error: Error in PushDeformer_deformGeometries. Missing attributes.
  1 kl.internal.String.SetErrorDataPtrAndLength.AS0()
  2 function.setError.in_AS0.ST()
  3 operator.PushDeformer_deformGeometries.in.UI32.io_AS0.OO_GeometrySet.in.F32() PushDeformer.kl:61
  4 internal.PushDeformer_deformGeometries.stub.cpu.00()
Error: GeometryStack/PushDeformer.kl:65:3: dereferenced null object
  1 kl.internal.String.SetErrorDataPtrAndLength.AS0()
  2 function.setError.in_AS0.ST()
  3 operator.PushDeformer_deformGeometries.in.UI32.io_AS0.OO_GeometrySet.in.F32() PushDeformer.kl:65
  4 internal.PushDeformer_deformGeometries.stub.cpu.00()
